modules/analysis-common/src/main/java/org/opensearch/analysis/common/CommonAnalysisModulePlugin.java

```diff
@@ -394,7 +394,21 @@ public Map<String, AnalysisProvider<TokenizerFactory>> getTokenizers() {
         // TODO deprecate and remove in API
         tokenizers.put("lowercase", XLowerCaseTokenizerFactory::new);
         tokenizers.put("path_hierarchy", PathHierarchyTokenizerFactory::new);
-        tokenizers.put("PathHierarchy", PathHierarchyTokenizerFactory::new);
+        tokenizers.put("PathHierarchy", (IndexSettings indexSettings, Environment environment, String name, Settings settings) -> {
+            if (indexSettings.getIndexVersionCreated().onOrAfter(Version.V_3_0_0)) {
```

Comment:
The deprecation comes in 3.0.0, right? (since we are now in 2.12 and it might be too aggressive to ask users to migrate). If yes, I don't think we need to thrown an exception but yes, log deprecation warning

Reply:
That is a good point. BTH I do not know how far the first 3.x release is.
It sounds to me that deprecation of `PathHierarchy` tokenizer name should have been deprecated at the same time as `nGrams` or `edgeGrams` but slipped under the radar. On one hand it is a breaking change so 3.x is a good opportunity to introduce it, on the other hand users should have enough time to prepare for this change.

I am open to any suggestions here.

Reply:
I would suggest to deprecate in 3.0.0 for removal in 4.0.0